### ISSUE TITLE
Check failed: initialized_ Create GPUMemory::Scope to initialize Memory Manager. In cpp application.

### ISSUE THREAD

**Issue by Smorodov** [NVIDIA/caffe#213]
"Check failed: initialized_ Create GPUMemory::Scope to initialize Memory Manager."

The above error raised when I try to run network constructor.
In both GPU and CPU modes.

(GTX 1070, Driver version 370, CUDA 8.0 (with patch), CUDNN 5.1, caffe 0.15, ubuntu 16.04)
It worked with earlier versions of caffe.

In DIGITS it works fine.

```
    if (useGPU)
        {
        Caffe::set_mode(Caffe::GPU);
        int device_id = 0;
        Caffe::SetDevice(device_id);
        cout << "Using GPU #" << device_id << endl;
        }
    else
        {
        cout << "Using CPU" << endl;
        Caffe::set_mode(Caffe::CPU);
        }

    net = new caffe::Net<float>(netArchitecture, caffe::TEST);  // <---- Error here.

```

Log in console:

I0821 01:25:23.858695 16936 net.cpp:52] Initializing net from parameters: 
state {
  phase: TEST
}
layer {
  name: "data"
  type: "MemoryData"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  memory_data_param {
    batch_size: 1
    channels: 1
    height: 64
    width: 64
  }
}
layer {
  name: "scale"
  type: "Power"
  bottom: "data"
  top: "scale"
  power_param {
    scale: 0.0125
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "scale"
  top: "conv1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 20
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 50
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "ip1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 500
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}
layer {
  name: "drop1"
  type: "Dropout"
  bottom: "ip1"
  top: "ip1"
  dropout_param {
    dropout_ratio: 0.8
  }
}
layer {
  name: "ip2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 6
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "prob"
  type: "Softmax"
  bottom: "ip2"
  top: "prob"
}
I0821 01:25:23.858876 16936 layer_factory.hpp:77] Creating layer data
I0821 01:25:23.858904 16936 net.cpp:94] Creating Layer data
I0821 01:25:23.858913 16936 net.cpp:409] data -> data
I0821 01:25:23.858954 16936 net.cpp:409] data -> label
I0821 01:25:23.877315 16936 net.cpp:144] Setting up data
I0821 01:25:23.877377 16936 net.cpp:151] Top shape: 1 1 64 64 (4096)
I0821 01:25:23.877388 16936 net.cpp:151] Top shape: 1 (1)
I0821 01:25:23.877393 16936 net.cpp:159] Memory required for data: 16388
I0821 01:25:23.877411 16936 layer_factory.hpp:77] Creating layer scale
I0821 01:25:23.877426 16936 net.cpp:94] Creating Layer scale
I0821 01:25:23.877434 16936 net.cpp:435] scale <- data
I0821 01:25:23.877465 16936 net.cpp:409] scale -> scale
I0821 01:25:23.877518 16936 net.cpp:144] Setting up scale
I0821 01:25:23.877526 16936 net.cpp:151] Top shape: 1 1 64 64 (4096)
I0821 01:25:23.877529 16936 net.cpp:159] Memory required for data: 32772
I0821 01:25:23.877533 16936 layer_factory.hpp:77] Creating layer conv1
I0821 01:25:23.877566 16936 net.cpp:94] Creating Layer conv1
I0821 01:25:23.877573 16936 net.cpp:435] conv1 <- scale
I0821 01:25:23.877580 16936 net.cpp:409] conv1 -> conv1
F0821 01:25:23.878811 16936 gpu_memory.cpp:68] Check failed: initialized_ Create GPUMemory::Scope to initialize Memory Manager
**\* Check failure stack trace: ***

**Comment by zazd** [comment]
net_.reset(new Net<float>(model_path, caffe::TEST, Caffe::GetDefaultDevice()));

try this. add Caffe::GetDefaultDevice()

**Comment by Smorodov** [comment]
It give error: ‘GetDefaultDevice’ is not a member of ‘caffe::Caffe’, where is it defined? 
I use [NVIDIA's branch 0.15 ](https://github.com/NVIDIA/caffe).

**Comment by zazd** [comment]
oh, sorry. Because I don't use this, I use other caffe branch.
maybe you can see https://github.com/NVIDIA/caffe/issues/196 if you have not seen it before.

**Comment by Smorodov** [comment]
Yes, I seen this. Thanks for reply anyway.

**Comment by jcAxe** [comment]
Did you solve this one? I am having basically the same message but at gpu_memory.cpp 74
`
F0328 16:06:39.022311   424 gpu_memory.cpp:74] Check failed: initialized_ Create GPUMemory::Scope to initialize Memory Manager`

**Comment by drnikolaev** [comment]
@keryar this is fixed in 0.16, please check it out

**Comment by gittigxuy** [comment]
@keryar did you solved this question,I meet the same question.Please help me

**Comment by Smorodov** [comment]
Resolved by moving to branch 0.16, later versions seems have no such problem.
